backend/src/main/kotlin/org/loculus/backend/controller/SubmissionController.kt

```diff
@@ -247,8 +250,16 @@ class SubmissionController(
                 schema = Schema(implementation = ProcessedData::class),
             ),
         ],
+        headers = [
+            Header(
+                name = "x-total-records",
+                description = "The total number of records sent in responseBody",
+                schema = Schema(type = "integer"),
+            ),
+        ],
     )
     @GetMapping("/get-released-data", produces = [MediaType.APPLICATION_NDJSON_VALUE])
+    @Transactional(isolation = Isolation.REPEATABLE_READ) // All operations will be performed on the same snapshot
```

Comment:
I feat that this won't work - the transaction where the stream happens is executed later. We only pass a callback to the output stream which will be executed somewhen later by Spring.

Reply:
Why do we actually not set our Isolation level to Serializable?

Reply:
What would be the benefit of that?

Reply:
Some information is also [here](https://en.wikipedia.org/wiki/Isolation_(database_systems)#Repeatable_reads) and [here](https://stackoverflow.com/questions/48417632/why-write-skew-can-happen-in-repeatable-reads).

One example:

> You are writing an application for doctors to manage their on-call shifts at a hospital. The hospital usually tries to have several doctors on call at any one time, but it absolutely must have at least one doctor on call. Doctors can give up their shifts (e.g., if they are sick themselves), provided that at least one colleague remains on call in that shift

```
BEGIN TRANSACTION;
    SELECT * FROM doctors
        WHERE on_call = true
        AND shift_id = 1234;
    if (current_on_call >= 2) {
        UPDATE doctors
        SET on_call = false WHERE name = 'Alice' AND shift_id = 1234;
    }
COMMIT;  
```

In repeatable read there is a race condition possible. I.e. the result will not always be the same as if the transactions were run one after the other, which is exactly what serializable guarantees

Reply:
I actually looked it up and Transactional uses the database's default, which is read-commited (one level lower than repeatable read) for Postgres